library/src/main/assets/leaflet_map.html

```diff
@@ -0,0 +1,502 @@
+      tms: false
+    },
+    Terrain: {
+      map: 'http://ditu.google.cn/maps/vt?pb=!1m5!1m4!1i{z}!2i{x}!3i{y}!4i256!2m3!1e4!2st!3i427!2m3!1e0!2sr!3i427129704!3m12!2sLANGTOKEN!3sREGIONTOKEN!5e18!12m4!1e68!2m2!1sset!2sTerrain!12m3!1e37!2m1!1ssmartmaps!4e0!5m1!5f2!23i1301875',
+      subdomains: [],
+      tms: false
+    },
+    Information: {
+      attribution: 'Map data &copy; Google'
+    }
+  }
+};
+
+L.CRS.Baidu = new L.Proj.CRS('EPSG:900913', '+proj=merc +a=6378206 +b=6356584.314245179 +lat_ts=0.0 +lon_0=0.0 +x_0=0 +y_0=0 +k=1.0 +units=m +nadgrids=@null +wktext  +no_defs', {
+  resolutions: function () {
+    var level = 19
+    var res = [];
+    res[0] = Math.pow(2, 18);
+    for (var i = 1; i < level; i++) {
+      res[i] = Math.pow(2, (18 - i))
+    }
+    return res;
+  }(),
+  origin: [0, 0],
+  bounds: L.bounds([20037508.342789244, 0], [0, 20037508.342789244])
+});
+
+L.tileLayer.provide = function(mapType) {
+  var type = 'MAPURL';
+
+  var layerProvider = L.tileLayer.providers[type][mapType];
+  if (layerProvider == null) {
+    return null;
+  }
+
+  var layer = L.tileLayer(layerProvider.map, {
+    attribution: L.tileLayer.providers[type].Information.attribution,
+    subdomains: layerProvider.subdomains,
+    tms: layerProvider.tms,
+    maxZoom: 18,
+    minZoom: 2,
+    scaler: window.devicePixelRatio,
+  });
+
+  return layer;
+};
+
+var markers = {};
+var polylines = {};
+var polygons = {};
+var moveTimeout;
+var mapOption = {
+  center: [23.16, 113.23],
+  zoom: 12,
+  zoomControl: false
+};
+var paddingLeft = 0;
+var paddingRight = 0;
+var paddingTop = 0;
+var paddingBottom = 0;
+var singleClick = false;
+if (isBaiduMap()) {
+  mapOption['crs']=L.CRS.Baidu;
+};
+var map = L.map('map', mapOption);
+var tileLayer = L.tileLayer.provide('Normal');
+tileLayer.addTo(map);
+AirMapView.onMapLoaded();
+
+map.on('click', function(event) {
+  singleClick = true;
+  setTimeout(function() {
+    if (singleClick) {
+      mapClick(event);
+      singleClick = false;
+    }
+  }, 200);
+});
+
+map.on('dblclick', function(event) {
+  singleClick = false;
+});
+
+map.on('moveend', function(event) {
+  mapMove();
+});
+
+function clearMarkers() {
+  for (lc = 0; lc < markers.length; lc++) {
+    map.removeLayer(markers[lc]);
+  }
+  markers = {};
+}
+
+function setBounds(neLat, neLng, swLat, swLng) {
+  var neLatlng = convertLatlng([ neLat, neLng ]);
+  var swLatlng = convertLatlng([ swLat, swLng ]);
+  var southWest = L.latLng(neLatlng[0], neLatlng[1]);
+  var northEast = L.latLng(swLatlng[0], swLatlng[1]);
+  var bounds = L.latLngBounds(southWest, northEast);
+  map.fitBounds(bounds);
+}
+
+function paddingLatLngFromLatLng(latLng) {
+  var centerPoint = map.latLngToContainerPoint(latLng);
+  centerPoint.x = centerPoint.x - (paddingLeft - paddingRight) / 2;
+  centerPoint.y = centerPoint.y - (paddingTop - paddingBottom) / 2;
+  return map.containerPointToLatLng(centerPoint);
+}
+
+function LatLngFromPaddingLatLng(paddingLatLng) {
```

Comment:
did you write this function or it was copied from somewhere else? JS functions should be camelCased

Reply:
Oh, it's a mistake. Thanks for pointing it out. I'll check all JS codes.